src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidator.java

```diff
@@ -263,4 +269,24 @@ private void validateFilingDate(LocalDate filingDate, Errors errors, String logg
             ApiLogger.infoContext(loggingContext, errorMessage);
         }
     }
+
+    private void validateHasSecureRegister(Boolean hasSecureRegister, boolean isFullRegistration, Errors errors, String loggingContext) {
+        if (!isRedisRemovalEnabled) {
+            return;
+        }
+
+        String qualifiedFieldName = OverseasEntitySubmissionDto.HAS_SECURE_REGISTER_FIELD;
+
+        if (isFullRegistration && hasSecureRegister == null) {
+            var errorMessage = String.format(ValidationMessages.NOT_NULL_ERROR_MESSAGE, qualifiedFieldName);
+            setErrorMsgToLocation(errors, qualifiedFieldName, errorMessage);
+            ApiLogger.infoContext(loggingContext, errorMessage);
+        }
+
+        if (Boolean.TRUE.equals(hasSecureRegister)) {
```

Comment:
If execution of the code has fallen into the error block above (lines 281 to 283 due to a `null` value) the value of `hasSecureRegister` can never be true. You could therefore safely avoid even doing this check by 'stitching' this on to the other `if` block and using an `else` statement.

Reply:
Yes. Personal preference. This is fine.